projects/core/src/lib/components/legend/services/legend.service.ts

```diff
@@ -63,6 +63,69 @@ export class LegendService {
       );
   }
 
+  public getLegendImages$(appLayers$: Observable<AppLayerWithServiceModel[]>, urlCallback?: (layer: AppLayerWithServiceModel, url: URL) => void):
+    Observable<Array<{ appLayer: AppLayerWithServiceModel; imageData: string; width: number; height: number }>> {
+    return this.getAppLayerAndUrl$(appLayers$).pipe(
+      concatMap(appLayerAndUrls => {
+        if (appLayerAndUrls.length === 0) {
+          return of([]);
+        }
+        return forkJoin(appLayerAndUrls.filter(lu => lu.url !== '').map(appLayerWithLegendUrl => {
+          const url = new URL(appLayerWithLegendUrl.url);
+          if (urlCallback) {
+            urlCallback(appLayerWithLegendUrl.layer, url);
+          }
+          return LegendService.imageUrlToPng$(url.toString()).pipe(
+            catchError((e) => {
+              console.log(`Error getting legend from URL ${appLayerWithLegendUrl.url}`, e);
+              return of(null);
+            }),
+            map(legendImage => ({
+                  ...legendImage,
+                  appLayer: appLayerWithLegendUrl.layer,
+            })),
+          );
+        }));
+      }),
+      map(array => {
+        return array.filter(a => a !== null) as Array<{ appLayer: AppLayerWithServiceModel; imageData: string; width: number; height: number }>;
```

Comment:
`... as ...` is something to prevent. This will effectively disable typechecking here so if something changes somewhere this is not picked up correctly. I supspect you can just just something like `array.filter(Boolean)` to keep Typescript happy. It can then infere the correct type automatically.